packages/block-editor/src/components/use-block-sync/index.js

```diff
@@ -210,8 +258,20 @@ export default function useBlockSync( {
 
 			if ( controlledSelectionStart && controlledSelectionEnd ) {
 				resetSelection(
-					controlledSelectionStart,
-					controlledSelectionEnd
+					shouldMapBlockIds()
+						? mapBlockId(
+								controlledSelectionStart,
+								instanceId,
+								DIRECTION_IN
+						  )
+						: controlledSelectionStart,
+					shouldMapBlockIds()
+						? mapBlockId(
+								controlledSelectionStart,
+								instanceId,
+								DIRECTION_IN
+						  )
+						: controlledSelectionEnd
```

Comment:
Another case where you could run `shouldMapBlockIds()` and check its value only once to improve performance.

Reply:
7cfc4ac